### PR TITLE
Documentation updates

### LINKED ISSUE #101: Typos in gsd.hoomd documentation
## Description

There are numerous typos in the gsd.hoomd documentation, such as BondData referring to the number of particles. We also need to switch to the preferred notation `list[str]` instead of using `Typing`.

## Developer

I will work on this at some point.

### PR DESCRIPTION
**Pull request by joaander**
## Description

Clean up typos and fix some wording in the documentation. Also provide some links to more details on HOOMD interprets the data in HOOMD's docs.

## Motivation and Context

The documentation should be accurate. This request fixes inaccurate or unclear statements.

Resolves #101

## How Has This Been Tested?

I viewed the documentation locally.

## Change log

```
Changed:

- Improved documentation.
```

## Checklist:

- [x] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/gsd/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**GSD Contributor Agreement**](https://github.com/glotzerlab/gsd/blob/master/ContributorAgreement.md).
- [x] My name is on the list of contributors (`doc/credits.rst`) in the pull request source branch.

### CODE REVIEW

File: INSTALLING.rst
Comment (joaander) [inline INSTALLING.rst:114]: I will take care of this TODO and remove it before merging.

File: gsd/pygsd.py
Comment (j-proc) [inline gsd/pygsd.py:11]: I personally liked the limits, but I do get that what's reasonable is extremely subjective.

File: gsd/fl.pyx
Comment (j-proc) [inline gsd/fl.pyx:691]: I'm still getting the `numpy.ndarray[type, ndim=?, mode='c']` return type when I built the docs. Is this meant to overwrite what I'm assuming is auto-generated by type hints.

File: doc/schema-hoomd.rst
Comment (j-proc) [inline doc/schema-hoomd.rst]: Good to have explicit 👍

Comment (j-proc) [reply doc/schema-hoomd.rst]: It's just a matter of preference, but I find it easier to read such ranges in the format 
```
-l_z/2` \le :math:`z < l_z/2`
```
I also wonder if this needs some reference back to the tilt factors for those who aren't familiar with them, the equations could be confusing. I'm not sure there's any notation like a hat that would make this obvious for novices.

File: doc/hoomd-examples.rst
Comment (j-proc) [inline doc/hoomd-examples.rst]: I don't think this one is accurate.

File: INSTALLING.rst
Comment (j-proc) [inline INSTALLING.rst]: It's nice to include these extra tips. I'm no sphinx expert so these were enlightening.

**Review by j-proc — COMMENTED** [review]
Nice documentation improvements. I left some comments and can approve if you think none need changing.

File: gsd/pygsd.py
Comment (joaander) [reply gsd/pygsd.py:11]: Yes, it is very subjective. Long term, I plan to deprecate and remove `pygsd`. The one use-case it allows is reading from an in-memory object but that is something we can add to the full gsd API #64.

With strong support on conda and pypi, users have no difficulty installing the full `gsd` package now.

File: gsd/fl.pyx
Comment (joaander) [reply gsd/fl.pyx:691]: Your locally generated docs may be pulling a gsd imported from some other installation. It is showing correctly here: https://gsd--118.org.readthedocs.build/en/118/python-module-gsd.fl.html#gsd.fl.GSDFile.read_chunk

Comment (joaander) [reply gsd/fl.pyx:691]: `freud` started using the "(N,M) ndarray of type" language a while ago and we have moved all the glotzer tools to this format. This was one lingering use of the previous format gsd used to document numpy array types.

File: doc/schema-hoomd.rst
Comment (joaander) [reply doc/schema-hoomd.rst]: Done. For better or worse, this is the same notation that LAMMPS uses: https://docs.lammps.org/Howto_triclinic.html

This is a schema reference so it doesn't need to be super friendly to new users. HOOMD's own documentation and tutorials does this. At some point, that will get improved Box documentation that will explain this better.

File: doc/hoomd-examples.rst
Comment (joaander) [reply doc/hoomd-examples.rst]: Fixed.

File: INSTALLING.rst
Comment (joaander) [reply INSTALLING.rst]: I'm moving away from the makefile that sphinx-quickstart generated and instead giving users the freedom to manually run sphinx-build (which many were doing anyways). However, the default options in sphinx leave a lot to be desired. If you have an import error, for example, sphinx will generate the docs and not display any errors or information to help you find the problem. It will happily produce documentation that is empty wherever autodoc failed to read the docstrings. These options help fix that. Also the `-a` option forces a rebuild of all files which is necessary when using the readthedocs sphinx theme. Otherwise partial rebuilds will result in inconsistent links in navigation pane.

**Review by j-proc — APPROVED** [review]

**Pull request merged**
